### DISCUSSION TITLE
legal issue about invest.com data scraping

### DISCUSSION THREAD

**Discussion by nhlsm** [alvarobartt/investpy#336] [Q&A] [answered]
Is it legal that scraping data from invest.com ?

I want to make android app with invest.com data.
I wonder it is legal.

**Comment by alvarobartt** [comment, marked as answer]
Yes it is, I asked them a couple of times and their response was basically that as long as you mention the data source where the data comes from (Investing.com) they are OK with it!

P.S. This is also why Investing.com is included in the project description and in the citation.

**Reply by nhlsm** [reply]
Thank you for super fast reply,
and good news.

**Comment by nhlsm** [comment]
one more question...

https://www.investing.com say that  ( bottom of page )

"...It is prohibited to use, store, reproduce, display, modify, transmit or distribute the data contained in this website **without the explicit prior written permission of Fusion Media and/or the data provider** ... "

------------------------------
I think it is different between "mention the data source where the data comes from"  and "the explicit prior written permission of Fusion Media and/or the data provider"

How do you think?

**Reply by alvarobartt** [reply]
Hi @nhlsm I think that as far as I understand you can use the data as I requested permission to develop `investpy` and people from Investing.com are aware of it, so you shouldn't have any problem. Anyway, feel free to contact the Investing.com team so as to ask them all the "legal doubts" you may have, as I don't use investpy in production, I just use it for research and to easily provide the data, but without any direct economic benefit from it.

**Reply by nhlsm** [reply]
Thanks you for super fast reply. ^^

**Reply by johnyhawkahsan** [reply]
@nhlsm is it possible to make Android application using investpy?
Sorry for the dumb questions but I was looking for a way to get historical data from investing.com because my country doesn't have any api available for stock market. I came across investpy but don't know how to use python based application to translate data on Android.

**Reply by nhlsm** [reply]
My intention is not use investpy directly on Android, but use ported version.

Android porting version has not been found so far, so it was intended to make sure there are no legal problems to implement it myself.

**Reply by alvarobartt** [reply]
AFAIK you shouldn't have problems using Investing.com data either through their web or via `investpy`, as long as you mention the source where the data comes from, in this case, a reference to the Investing.com web is enough I think.

If you want more details, feel free to contact customer support on Investing.com! 👍🏻
